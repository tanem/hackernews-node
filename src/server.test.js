const request = require('supertest')
const server = require('./server').server
const prisma = require('./server').prisma

let httpServer

beforeAll(async () => {
  httpServer = await server.start()
})

afterAll(async () => {
  httpServer.close()
  await prisma.disconnect()
})

it('should return feed', async () => {
  const { body, status } = await request(httpServer).post('/').send({
    query: `
      {
        feed {
          count
          links {
            id
          }
        }
      }
    `,
  })

  expect(status).toBe(200)
  expect(body).toMatchSnapshot()
})
