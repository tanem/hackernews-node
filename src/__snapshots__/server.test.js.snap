// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return feed 1`] = `
Object {
  "data": Object {
    "feed": Object {
      "count": 5,
      "links": Array [
        Object {
          "id": "1",
        },
        Object {
          "id": "2",
        },
        Object {
          "id": "3",
        },
        Object {
          "id": "4",
        },
        Object {
          "id": "5",
        },
      ],
    },
  },
}
`;
